server/routes/auth.js

```diff
@@ -0,0 +1,108 @@
+import express from 'express'
+import bcrypt from 'bcrypt'
+import jwt from 'jsonwebtoken'
+import User from '../models/userModel'
+
+const router = express.Router()
+
+//Middleware validation cho API đăng ký
+const validateSignup = (req, res) => {
+    const { name, email, phone, address, password, confirmPassword } = req.body
+
+    //Kiểm tra các trường có giá trị không
+    if (!name || !email || !phone || !address || !password || !confirmPassword) {
+        return res.status(400).json({ message: 'Vui lòng điền đầy đủ thông tin' })
+    }
+    
+    // Kiểm tra định dạng email
+    if (!email.match(/^[^\s@]+@[^\s@]+\.[^\s@]+$/)) {
```

Comment:
I think you should seperate this code as 
```suggestion
    const isPhoneNumberValid = (phone) => /^\d{10,11}$/.test(phone);
const isEmailValid = (email) => /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
```
After you call function in here